### PR TITLE
Fix reentrant remove behavior

### LINKED ISSUE #18537: Behavior removal bug in `MultiPointerScaleBehavior.detach()`
`MultiPointerScaleBehavior` fails to remove itself correctly when `Node.removeBehavior()` is called, possibly event removing another, unrelated behaviour. Later this raises an exception since the addition of `this._ownerNode = null!;` in `MultiPointerScaleBehavior.detach()`.

This relates directly to this [forum post](https://forum.babylonjs.com/t/removing-a-behavior-inside-another-behaviors-detach-breaks-removebehavior/63549)

**Repro**

This playground sample adds the `MultiPointerScaleBehavior`, prints the number and names of the added behaviours, removes the `MultiPointerScaleBehavior` and does the same again. This should result in no behaviours left, but leaves one active.

- Bug repro on [playground](https://playground.babylonjs.com): https://playground.babylonjs.com/#M9DMC0 (vibe coded example)
- Expected result: "After removeBehavior() - 0 behavior(s): []"
- Current result: "After removeBehavior() - 1 behafior(s): ["MultiPointerScale"]"

**Desktop:**
 - OS: Linux (all)
 - Browser: Edge (all)
 - Version: 9.10.1

**Additional context**

The issue is that `Node.removeBehavior()` follows this sequence of events (assuming `MultiPointerScaleBehavior` is the only registered behaviour for illustrative purposes):

1. Get the index of the `MultiPointerScaleBehavior` => 2
2. Call `MultiPointerScaleBehavior.detach()`
3. Call `Node.removeBehavior()` on the two child behaviours
    - Removes behaviours at indices {0, 1}
4. Tries to remove the `MultiPointerScaleBehavior` from `Node._behaviors` using the index from step 1 (2).

The index used at step 4 is now stale and will either do nothing (best case) or remove a different behaviour.

Possible fix: effect step 4 before step 3 -

```ts
    public removeBehavior(behavior: Behavior<Node>): Node {
        const index = this._behaviors.indexOf(behavior);

        if (index === -1) {
            return this;
        }

        // These two lines have been swapped
        this._behaviors.splice(index, 1);
        this._behaviors[index].detach();

        return this;
    }
```

### PR DESCRIPTION
**Pull request by CedricGuillemet**
fixes #18537

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/18552/merge

Test environment:
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18552/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18552/merge/index.html#WGZLGJ#4600

Links to test your changes to core in the published versions of the Babylon tools (does not contain changes you made to the tools themselves):

https://playground.babylonjs.com/?snapshot=refs/pull/18552/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/18552/merge
https://gui.babylonjs.com/?snapshot=refs/pull/18552/merge
https://nme.babylonjs.com/?snapshot=refs/pull/18552/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/18552/merge#BCU1XR#0

*If you made changes to the sandbox or playground in this PR, additional comments will be generated soon containing links to the dev versions of those tools.*

**Comment by bjsplat** [comment]
Interaction tests

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18552/merge/testResults/interactionplaywright/index.html

**Review by sebavan — APPROVED** [review]

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18552/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
## 🟢 Memory Leak Test Results

**13** passed, **0** leaked out of **13** scenarios

🟢 All memory leak tests passed — no leaks detected.

<details>
<summary>Passed Scenarios (13)</summary>

| Scenario | Package |
|----------|---------|
| Core Feature Stack | @babylonjs/core |
| Core Rendering Materials Shadows Stack | @babylonjs/core |
| Core Textures Render Targets PostProcess Stack | @babylonjs/core |
| GUI Fullscreen UI Controls | @babylonjs/gui |
| GUI Mesh ADT Controls | @babylonjs/gui |
| Loaders Boombox Import | @babylonjs/loaders |
| Loaders OBJ Direct Load | @babylonjs/loaders |
| Loaders STL Direct Load | @babylonjs/loaders |
| Materials Library Stack | @babylonjs/materials |
| Serializers glTF Export | @babylonjs/serializers |
| Serializers GLB Export | @babylonjs/serializers |
| PostProcesses Digital Rain Stack | @babylonjs/post-processes |
| Procedural Textures Stack | @babylonjs/procedural-textures |

</details>

**Comment by bjsplat** [comment]
Visualization tests for WebGPU

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18552/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
## ⚡ Performance Test Results

🟢 All performance tests passed — no regressions detected.

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18552/merge/testResults/webgl2playwright/index.html
